src/inode.rs

```diff
@@ -74,6 +74,25 @@ impl Inode {
 			flags: self.flags,
 		}
 	}
+
+	pub fn size(&self, bs: u64, fs: u64) -> (u64, u64) {
+		let size = match self.kind() {
+			FileType::Directory => self.blocks * fs,
+			FileType::RegularFile | FileType::Symlink => self.size,
+			kind => todo!("Inode::size() is undefined for {kind:?}"),
+		};
+		Self::inode_size(bs, fs, size)
+	}
+
+	/// The number of blocks and fragments this inode needs.
+	fn inode_size(bs: u64, fs: u64, size: u64) -> (u64, u64) {
+		let mut x = size;
+		let blocks = x / bs;
+		x -= bs * blocks;
+		let frags = (x + fs - 1) / fs;
```

Comment:
You can simplify this as:
```suggestion
		let blocks = size / bs;
		let frags = (size % bs).div_ceil(fs);
```